types/jsonld/jsonld.d.ts

```diff
@@ -0,0 +1,254 @@
+        | IncludedBlock
+        | IdMap
+        | TypeMap
+        | NodeObject[keyof NodeObject];
+}
+
+/**
+ * A graph object represents a named graph, which MAY include an explicit graph name.
+ * @see https://www.w3.org/TR/json-ld11/#graph-objects
+ */
+export interface GraphObject {
+    '@graph': OrArray<NodeObject>;
+    '@index'?: Keyword['@index'];
+    '@id'?: Keyword['@id'];
+    '@context'?: Keyword['@context'];
+}
+
+/**
+ * A value object is used to explicitly associate a type or a language with a value
+ * to create a typed value or a language-tagged string and possibly associate a base direction.
+ * @see https://www.w3.org/TR/json-ld11/#value-objects
+ */
+export type ValueObject = {
+    '@index'?: Keyword['@index'];
+    '@context'?: Keyword['@context'];
+} & ({
+    '@value': Keyword['@value'];
+    '@language'?: Keyword['@language'];
+    '@direction'?: Keyword['@direction'];
+} | {
+    '@value': Keyword['@value'];
+    '@type': Keyword['@type'];
+} | {
+    '@value': Keyword['@value'] | object | any[];
+    '@type': '@json';
+});
+
+/**
+ * A list represents an ordered set of values.
+ * @see https://www.w3.org/TR/json-ld11/#lists-and-sets
+ */
+export interface ListObject {
+    '@list': Keyword['@list'];
+    '@index'?: Keyword['@index'];
+}
+
+/**
+ * A set represents an unordered set of values.
+ * @see https://www.w3.org/TR/json-ld11/#lists-and-sets
+ */
+export interface SetObject {
+    '@set': Keyword['@set'];
+    '@index'?: Keyword['@index'];
+}
+
+/**
+ * A language map is used to associate a language with a value in a way that allows easy programmatic access.
+ * @see https://www.w3.org/TR/json-ld11/#language-maps
+ */
+export interface LanguageMap {
+    [key: string]: null | string | string[];
+}
+
+/**
+ * An index map allows keys that have no semantic meaning, but should be preserved regardless,
+ * to be used in JSON-LD documents.
+ * @see https://www.w3.org/TR/json-ld11/#index-maps
+ */
+export interface IndexMap {
+    [key: string]: OrArray<
+        | null
+        | boolean
+        | number
+        | string
+        | NodeObject
+        | ValueObject
+        | ListObject
+        | SetObject
+    >;
+}
+
+/**
+ * An id map is used to associate an IRI with a value that allows easy programmatic access.
+ * @see https://www.w3.org/TR/json-ld11/#id-maps
+ */
+export interface IdMap {
+    [key: string]: NodeObject;
+}
+
+/**
+ * A type map is used to associate an IRI with a value that allows easy programmatic access.
+ * @see https://www.w3.org/TR/json-ld11/#type-maps
+ */
+export interface TypeMap {
+    [key: string]: string | NodeObject;
+}
+
+/**
+ * An included block is used to provide a set of node objects.
+ * @see https://www.w3.org/TR/json-ld11/#included-blocks
+ */
+export type IncludedBlock = OrArray<NodeObject>;
+
+/**
+ * A context definition defines a local context in a node object.
+ * @see https://www.w3.org/TR/json-ld11/#context-definitions
+ */
+export interface ContextDefinition {
```

Comment:
since this is type-only, could you keep the name the same as before? Otherwise all the existing dependents have to change (as in rdjs__parser below)

Reply:
Yes, I thought that might be a problem. The only breaking change I made was that I [removed](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/47364/files#diff-faf773fd4d0109e1c8cd2339c8acdb2e421f1c7072b426e18c067097bf84eb2fL18) the export of `Context` and `Document` (which aliased to `JsonLd|Url` and `Document` respectively) from `json-spec.d.ts`, and doing that required me to update the dependents as you noted.

I would prefer to keep the type names as close as possible to the [spec](https://www.w3.org/TR/json-ld11/), so maybe we can find a compromise. If I could re-add `Context` and `Document` back into `json-spec.d.ts`, but change their aliases from the old `JsonLd|Url` and `Document` to the new `ContextDefinition` and `JsonLdDocument` respectively, then I believe that would make the change non-breaking as well as keep the same names as before.

Reply:
That's a good idea. 👍🏼